robocop/checkers/naming.py

```diff
@@ -679,7 +699,9 @@ def visit_KeywordCall(self, node):  # noqa
             if len(node.data_tokens) < 2:
                 return
             token = node.data_tokens[1]
-            if token.type == Token.ARGUMENT and not token.value.isupper():
+            var_name = search_variable(token.value).base
+            normalized_var_name = remove_nested_variables(var_name)
+            if token.type == Token.ARGUMENT and not normalized_var_name.isupper():
```

Comment:
Same as previous comment - if token.type != Token.ARGUMENT, we will not need var_name and normalized_var_name . In that case we could return early